pymc/distributions/distribution.py

```diff
@@ -355,205 +363,66 @@ def dist(
-        """
-
-        try:
-            from pymc.model import Model
+class SymbolicRandomVariable(OpFromGraph):
+    """Symbolic Random Variable
 
-            model = Model.get_context()
-        except TypeError:
-            raise TypeError(
-                "No model on context stack, which is needed to "
-                "instantiate distributions. Add variable inside "
-                "a 'with model:' block, or use the '.dist' syntax "
-                "for a standalone distribution."
-            )
+    This is a subclasse of `OpFromGraph` which is used to encapsulate the symbolic
+    random graph of complex distributions which are built on top of pure
+    `RandomVariable`s.
 
-        if "testval" in kwargs:
-            initval = kwargs.pop("testval")
-            warnings.warn(
-                "The `testval` argument is deprecated; use `initval`.",
-                FutureWarning,
-                stacklevel=2,
-            )
+    `ndim_supp` must be specified when a variable is created.
 
-        if not isinstance(name, string_types):
-            raise TypeError(f"Name needs to be a string but got: {name}")
+    A logprob function can be dispatched directly to subclasses. Otherwise, it is
+    possible to rely on Aeppl to derive the logprob by introspection of the inner graph.
+    To do this set the flag `inline_aeppl` to True (default is False).
 
-        dims = convert_dims(dims)
-        if observed is not None:
-            observed = convert_observed_data(observed)
-
-        # Create the RV, without taking `dims` into consideration
-        rv_out, resize_shape_from_dims = _make_rv_and_resize_shape_from_dims(
-            cls=cls, dims=dims, model=model, observed=observed, args=args, **kwargs
-        )
+    The `_print_name` attribute can be overriden by sublclasses for pretty printing.
 
-        # Resize variable based on `dims` information
-        if resize_shape_from_dims:
-            resize_size_from_dims = find_size(
-                shape=resize_shape_from_dims, size=None, ndim_supp=rv_out.tag.ndim_supp
-            )
-            rv_out = cls.change_size(rv=rv_out, new_size=resize_size_from_dims, expand=False)
+    SymbolicRandomVariable may implement an `update` method which returns a dictionary
+    with the symbolic expressions required for correct random state updating between
+    repeated function evaluations. This is used by `aesaraf.compile_pymc`.
+    """
 
-        rv_out = model.register_rv(
-            rv_out,
-            name,
-            observed,
-            total_size,
-            dims=dims,
-            transform=transform,
-            initval=initval,
-        )
-        # add in pretty-printing support
-        rv_out.str_repr = types.MethodType(str_for_symbolic_dist, rv_out)
-        rv_out._repr_latex_ = types.MethodType(
-            functools.partial(str_for_symbolic_dist, formatting="latex"), rv_out
-        )
+    ndim_supp: int = None
+    inline_aeppl: bool = False
```

Comment:
```suggestion
    inline_aeppl: bool = False
    """Specifies whether the logprob function is derived automatically by introspection of the inner graph.
    
    If `False`, a logprob function must be dispatched directly to the subclass.
    """
```

It would help to have such docstrings on the other attributes too..

Reply:
I already included the explanation on the docstrings above. Is there a standard for explaining attributes, like the `parameters` in numpy docstrings?

Reply:
I'm not sure if I understand the question correctly.

You can put docstrings on attributes by placing a `""" ... """` below them. That's the standard, it's just not well known..

Reply:
Didn't know. Done